### ISSUE TITLE
Pandas bar plot only shows axes and ticks

### ISSUE THREAD

**Issue by jcommelin** [nschloe/matplotlib2tikz#84]
It seems that a basic bar plot by Pandas does not correctly render the bars.

``` python
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib2tikz import save as tikzsave

s = pd.Series({'a':3, 'b':5, 'c':2})
s.plot(kind="bar")
tikzsave("testplot.tex")
```

This generates a tex file with 12 lines of tex (not counting comments).
The 12 lines setup the axes of the plots, and put labels ('a', 'b', and 'c') on the axis. But the actual bars (with values 3, 5, and 2) are not drawn.

**Comment by andrewbolster** [comment]
:+1:

**Comment by andrewbolster** [comment]
Little bit more investigation I may as well dump here before I give up and hand craft this madness; basically it looks like mpl2tkz simply can't tell the difference between an arbitrary `ax.patch` and a actual bar chart, so the bar plots don't work in the conventional `pgfplots`way. In particular, legends just don't work as there's no recorded "plot" to label.

Even with a minimal native example;

```
import matplotlib.pyplot as plt
from matplotlib2tikz import save as tikzsave

plt.bar(range(5), [np.random.randint(10) for _ in range(5)])
tikzsave("testplot.tex", draw_rectangles=True)
```
Gives 
```
\begin{tikzpicture}

\definecolor{color1}{rgb}{0.298039215686275,0.447058823529412,0.690196078431373}
\definecolor{color0}{rgb}{0.917647058823529,0.917647058823529,0.949019607843137}

\draw[opacity=0] (axis cs:0,0) rectangle (axis cs:1,1);
\begin{axis}[
xmin=0, xmax=5,
ymin=0, ymax=9,
tick align=outside,
xmajorgrids,
x grid style={white},
ymajorgrids,
y grid style={white},
axis line style={white},
axis background/.style={fill=color0}
]
\draw[draw=black,fill=color1] (axis cs:0,0) rectangle (axis cs:0.8,9);
\draw[draw=black,fill=color1] (axis cs:1,0) rectangle (axis cs:1.8,1);
\draw[draw=black,fill=color1] (axis cs:2,0) rectangle (axis cs:2.8,6);
\draw[draw=black,fill=color1] (axis cs:3,0) rectangle (axis cs:3.8,6);
\draw[draw=black,fill=color1] (axis cs:4,0) rectangle (axis cs:4.8,6);
\draw[fill=color0,draw opacity=0] (axis cs:0,0) rectangle (axis cs:1,1);
\end{axis}

\end{tikzpicture}% 
```
(without the `draw_rectangles` argument, these draws don't appear).

Also noteworthy, is that in order to compile anything you need to kill the last `\draw[fill=color0,draw opacity=0] (axis cs:0,0) rectangle (axis cs:1,1);' line. I'm no expert but it looks like it's trying to draw the background, but it errors out for me with "Missing \endcsname inserted."

Hope that's helpful to someone. 

TLDR:
1. Pandas isn't the problem; plotting bars is.
1. additional rectangles break things
1. Broken Bars break legends.

**Comment by nschloe** [comment]
This is now fixed. For
```python
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib2tikz import save as tikzsave

s = pd.Series({'a':3, 'b':5, 'c':2})
s.plot(kind="bar")
tikzsave("testplot.tex")
```
I'm getting
![a](https://cloud.githubusercontent.com/assets/181628/22607450/41e0a670-ea58-11e6-8e5c-5822cdd16222.png)
